tests/functional/aws-node-sdk/test/bucket/testBucketStress.js

```diff
@@ -0,0 +1,55 @@
+import { S3 } from 'aws-sdk';
+import { times, timesSeries, waterfall } from 'async';
+
+import getConfig from '../support/config';
+
+const bucket = `stress-test-bucket-${Date.now()}`;
+const text = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+const objectCount = 100;
+const loopCount = 10;
+
+function putObjects(s3, cb) {
+    let count = 0;
+    times(objectCount, (i, next) => {
+        const params = { Bucket: bucket, Key: `foo${i}`, Body: text };
+        s3.putObject(params, error => {
+            if (error || ++count === objectCount) {
+                cb(error);
```

Comment:
Ah, I misread this.  I thought you were trying to catch a counting error.  You don't have to do all of this work with async.times.  It will keep track of where you are in the loop for you.  You can just call next(error) and then your last argument to the times function is the cb which gets called immediately if there is an error or at the end when you are done.